sympy/functions/elementary/piecewise.py

```diff
@@ -806,6 +807,25 @@ def _eval_nseries(self, x, n, logx, cdir=0):
         args = [(ec.expr._eval_nseries(x, n, logx), ec.cond) for ec in self.args]
         return self.func(*args)
 
+    def _eval_factor(self):
+        from sympy.polys import factor, cancel
+        from sympy.polys.polyerrors import PolynomialError
+
+        args = [(e if hasattr(e, "_eval_factor") else factor(e), c) for e, c in self.args]
+        commonfact = None
+        for expr, cond in args:
+            if commonfact is not None:
+                try:
+                    commonfact = gcd(commonfact, expr)
+                except PolynomialError:
+                    commonfact = 1
+            else:
+                commonfact = expr
+        if commonfact == 1 and args == self.args:
```

Comment:
Why do you have `args == self.args` here?

What if some of the arguments are modified by `factor` and `commonfact` is still 1? In that case you will divide by one and factor out a one which doesn't make sense (or am I missing something?).

Reply:
You're right, dividing and factoring by 1 is not very efficient. There is only one case where you want factor to factor Piecewise arguments, and that is when one of those arguments is a gamma expression, because "gammasimp" counts on that.